src/managers/MatchesManager.js

```diff
@@ -0,0 +1,131 @@
+const ReqUtil = require('../util/ReqUtil');
+const CacheManager = require('./CacheManager');
+const {MMTypes, MatchmakingGroup, MatchStatus} = require('../util/Constants'); // eslint-disable-line no-unused-vars
+const Match = require('../structures/Match');
+const Client = require('../client/Client'); // eslint-disable-line no-unused-vars
+
+/**
+ * Represents the matches manager (3v3 or Royal matches).
+ */
+class MatchesManager{
+    constructor(client){
+        /**
+         * The client instance.
+         * @type {Client}
+         * @readonly
+         */
+        this.client = client;
+
+        /**
+         * The cache manager
+         * @type {CacheManager}
+         * @private
+         */
+        this._cache = new CacheManager(this.client, this, Match);
+    }
+
+    /**
+     * Get a list of the matches
+     * @param {MatchmakingGroup} [group='3v3'] The group to get the matches from
+     * @param {number} [page=0] The page to get
+     * @param {Boolean} [cache=this.client.options.cache.enabled] Whether to cache the matches or not
+     * @returns {Promise<Array<MatchResult>>}
+     */
+    async list(group = 2, page = 0, cache = this.client.options.cache.enabled){
+        if (typeof group == 'string') group = MMTypes[group];
+        const matches = this.client.options.api.paths.tmio.tabs.matches;
+        const res = await this.client._apiReq(`${new ReqUtil(this.client).tmioAPIURL}/${matches}/${group}/${page}`);
+        const array = [];
+        if (res.matches.length > 0) { // check all news from the page 0
+            for (let i = 0; i < res.matches.length; i++) {
+                let match = new MatchResult(this.client, res.matches[i]);
+                if (cache) {
+                    res.matches[i]._cachedTimestamp = Date.now();
+                    this._cache.set(res.id, match);
+                }
+                array.push(match);
+            }
+        }
+        return array;
+    }
+
+    /**
+     * Fetches a Trackmania Match and returns its data.
+     * @param {string} liveID The live ID of the match (should start with 'LID-MTCH')
+     * @param {boolean} [cache=this.client.options.cache.enabled] Whether to get the news from cache or not
+     * @returns {Promise<Match>} The Match
+     */
+    async get(liveID, cache = this.client.options.cache.enabled){
+        if (cache && this._cache.has(liveID)) {
+            return this._cache.get(liveID);
+        } else {
+            return await this._fetch(liveID, cache);
+        }
+    }
+
+    /**
+     * Fetches a Match and returns its data
+     * @param {number} liveID Thelive ID of the match (should start with 'LID-MTCH')
```

Comment:
1. If the `liveID` should start with `"LID-MTCH"`, how can it be of type `number`?
2. Not 100% sure if there are any edge cases - but if it always starts with that prefix, it could be worth checking for that and throwing an appropriate error for other values, just to make it easier for the user of the library.